sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -1155,16 +1158,36 @@ <T> long insertAll(
                     .setErrors(ImmutableList.of(new ErrorProto().setReason("row-too-large")));
             // We verify whether the retryPolicy parameter expects us to retry. If it does, then
             // it will return true. Otherwise it will return false.
-            Boolean isRetry = retryPolicy.shouldRetry(new InsertRetryPolicy.Context(error));
-            if (isRetry) {
+            if (retryPolicy.shouldRetry(new InsertRetryPolicy.Context(error))) {
+              // Create row details composed of key value pairs.
+              String rowDetails;
```

Comment:
Does logging the row details help resolve https://github.com/apache/beam/issues/35566 since it mentions the underlying schema is actually wrong? I think we should somehow surface the schema issue here, right?

And in general, we do not want to log the raw rows since the data could be sensitive.

Reply:
Updated to compare the rows to the schema and note what are the differences.